lib/bundles/inspec-compliance/api.rb

```diff
@@ -204,11 +204,22 @@ def self.is_compliance_server?(config)
     end
 
     def self.is_automate_server_pre_080?(config)
-      config['server_type'] == 'automate' && config['version'].nil?
+      # Automate versions before 0.8.x may have a "version" key in the config.
+      # Unless it's a hash that also contains a "version" key, it came from
+      # an Automate server that is pre-0.8.x.
+      return false unless config['server_type'] == 'automate'
+      return true unless config.key?('version')
+      return true unless config['version'].is_a?(Hash)
+      config['version']['version'].nil?
     end
 
     def self.is_automate_server_080_and_later?(config)
-      config['server_type'] == 'automate' && !config['version'].nil?
```

Comment:
I don't think that's necessary. I'd rather these methods be able to stand on their own rather than depending on themselves. It also makes it clearer to the next person that stumbles upon this part of the code.